src/rules/fast-deep-equal-over-microdiff.ts

```
@@ -28,61 +28,246 @@ export const fastDeepEqualOverMicrodiff = createRule<[], MessageIds>({
+    defNode.init.callee.name === microdiffImportName
+  ) {
+    return defNode.init as TSESTree.CallExpression;
+  }
+  return undefined;
+}
+
+/**
+ * Determine if the given variable is used only for `.length` property accesses.
+ * If there is any non-length usage (e.g., indexing, iteration, method calls), return false.
+ */
+function isVariableOnlyUsedForLength(identifier: TSESTree.Identifier): boolean {
```

ESLint

[error] Replace `function·isVariableOnlyUsedForLength(identifier:·TSESTree.Identifier` with `····function·isVariableOnlyUsedForLength(⏎······identifier:·TSESTree.Identifier,⏎····`

(prettier/prettier)